### PR TITLE
Update dependencies to recent versions (imgui ~1.89.3, etc.)

### PR DESCRIPTION
**Pull request by jminor**
This PR updates most of the libraries in `libs` to more recent versions.
This PR also adds some additional instructions about build dependencies to the README and fixes the build status badge.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -129,8 +129,8 @@ jobs:
         mkdir build_win64
         cd build_win64
         cmake ..
-        cmake --build . --config Debug -j
-        cmake --build . --config Release -j
+        cmake --build . --config Debug -j --parallel 4
```
Comment (rogernelson) [inline .github/workflows/build.yaml:132]: Don't `-j` and `--parallel 4` conflict here? 

From the cmake --build help:
>   --parallel [<jobs>], -j [<jobs>]
                 = Build in parallel using the given number of jobs. 
                   If <jobs> is omitted the native build tool's 
                   default number is used.
                   The CMAKE_BUILD_PARALLEL_LEVEL environment variable
                   specifies a default parallel level when this option
                   is not given.

Not sure what will happen if both are specified with different values.

**Review by rogernelson — APPROVED** [review]
Looks good to me!

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -18,14 +18,14 @@ jobs:
     - name: Install Dependencies
       run: |
         sudo apt-get update
-        sudo apt-get install -y libglfw3-dev libsdl2-dev libgtk-3-dev
+        sudo apt-get install -y libglfw3-dev libgtk-3-dev
```
Comment (meshula) [inline .github/workflows/build.yaml:21]: The SDL2 dependency was for emsc I think. Later the comments say refer to the build.yaml to learn how to build, do you need an extra comment that indicates sdl2 is an additional dependency?

File: README.md [hunk 3]
```diff
@@ -20,6 +36,8 @@ This tool aims to replace [otioview](https://github.com/AcademySoftwareFoundatio
 
 ## Building (WASM via Emscripten)
 
+You will need to install the [Emscripten toolchain](https://emscripten.org) first.
+
```
Comment (meshula) [inline README.md:40]: I think you might mention sdl2 here? Or does the emsc toolchain come with sdl2?

**Review by meshula — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (JeanChristopheMorinPerso) [reply README.md:40]: See https://emscripten.org/docs/compiling/Building-Projects.html?highlight=sdl2#emscripten-ports.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (jminor) [reply .github/workflows/build.yaml:132]: Good catch. Let me try with just one to see what happens.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (jminor) [reply .github/workflows/build.yaml:21]: In a previous iteration of Raven I was using the SDL backend, but later switched to the GLFW backend, which made this install irrelevant.

File: README.md (hunk 3 above)
Comment (jminor) [reply README.md:40]: Yes, SDL2 is included with Emscripten.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (jminor) [reply .github/workflows/build.yaml]: Yep, that worked.

**Pull request merged**